### ISSUE TITLE
plotwitherror might omit error bars in log scale

### ISSUE THREAD

**Issue by NikSchlage** [HISKP-LQCD/hadron#158]
Trying to plot the data in [strange_data.txt](https://github.com/HISKP-LQCD/hadron/files/2868770/strange_data.txt) using

```r
with(df, plotwitherror(x, y, dy, log = 'y'))
```

results in a plot where no error bars are shown:

![corr1t96l24](https://user-images.githubusercontent.com/41633300/52849704-f9ffb280-3111-11e9-85a7-0202b98be2a8.png)

Without the `log = 'y'` parameter, the error bars are shown.

With ggplot the error bars are visible in log-scale:

```r
ggplot(df, aes(x = x, y = y)) +
  geom_point() +
  geom_errorbar(aes(ymin = y - dy, ymax = y + dy)) +
  scale_y_log10() +
  theme_light()
```

![corr1t96l24_2](https://user-images.githubusercontent.com/41633300/52849960-9cb83100-3112-11e9-908f-ca066f89dd50.png)

So the data seems legitimate, but somehow `plotwitherror` does not show the errors.

**Comment by kostrzewa** [comment]
This is due to fdc169c315fe7f651916728c21827bf8419c3c23 by @j-ostmeyer . The errors are so small that plotwitherror skips them. Incorrectly, it skips all of them. Reverting that commit makes things work as they should again.

**Comment by kostrzewa** [comment]
Sorry, had the wrong commit id there...

**Comment by kostrzewa** [comment]
@j-ostmeyer if you can find a better solution, that would be great. If not, I'm afraid we should probably revert your commit.

**Comment by j-ostmeyer** [comment]
Thanks for the remark. I would not have thought that the error is there.

**Comment by j-ostmeyer** [comment]
Solved by #160 .

**Comment by kostrzewa** [comment]
@NikSchlage can you confirm that #160 solves this for you?

**Comment by NikSchlage** [comment]
Yes, thank you all. Now the problem is solved and the error bars are shown:

![corr1_t96l24c](https://user-images.githubusercontent.com/41633300/52869291-a0b17680-3145-11e9-8ba2-4241c68a6882.png)

**Comment by kostrzewa** [comment]
Okay, still looks a bit odd as a few error bars are missing, but I think we might just accept this, agreed @martin-ueding @urbach?

**Comment by j-ostmeyer** [comment]
The error bars (actually only the lower half of them) are missing because `y-dy < 0`. One could argue that this is not the expected behaviour, but I think that it is a good solution and I omitted these lower halfs on purpose.

**Comment by kostrzewa** [comment]
> The error bars (actually only the lower half of them) are missing because `y-dy < 0`. One could argue that this is not the expected behaviour, but I think that it is a good solution and I omitted these lower halfs on purpose.

That part is fine. I mean the first few points for the smallest source-sink separations. It's clear that here the 1/1000 condition is triggered and one can argue that we should not draw anything there, but it does look a bit surprising.

**Comment by j-ostmeyer** [comment]
Ok, we can just draw a line there if the arrows fail. Is that what you suggest?

**Comment by kostrzewa** [comment]
> Ok, we can just draw a line there if the arrows fail. Is that what you suggest?

Sounds like a good idea!

**Comment by kostrzewa** [comment]
That way zooms will also work as expected.

**Comment by j-ostmeyer** [comment]
> That way zooms will also work as expected.

Not quite. R is a funny language. Arrow heads are measured in inches, but lines are drawn between coordinates. That means if I zoom in, the arrow heads stay the same size and lines grow. Do you have an idea how to make this consistent?
I will add a version using `lines` to #160 anyway.

**Comment by j-ostmeyer** [comment]
A different option would be to set the arrow length to 1/1000 inch if it would be shorter otherwise.

**Comment by kostrzewa** [comment]
> Not quite. R is a funny language. Arrow heads are measured in inches, but lines are drawn between coordinates. That means if I zoom in, the arrow heads stay the same size and lines grow. Do you have an idea how to make this consistent?

The arrow heads should stay the same size, I would not want the heads of error bars taking up 10% of the plot width each at high zoom levels.

**Comment by j-ostmeyer** [comment]
I agree. So I will set all arrows to `length >= 1/1000 inch`.

**Comment by kostrzewa** [comment]
Resolved via #160
